python/ray/serve/_private/deployment_state.py

```diff
@@ -1391,6 +1400,17 @@ def deployment_name(self) -> str:
     def app_name(self) -> str:
         return self._id.app_name
 
+    @property
+    def docs_path(self) -> Optional[str]:
+        replicas = self._replicas.get([ReplicaState.RUNNING])
+        docs_paths = [replica.docs_path for replica in replicas]
+        if len(set(docs_paths)) > 1:
+            raise ValueError(
+                f"All replicas must have the same docs path. "
+                f"Got {docs_paths} for deployment {self._id}."
+            )
```

Comment:
This will trigger a value error if there are replicas of different versions running, so we can't fail on that. You should also filter by `version=target_version` when fetching replicas.